src/optics.jl

```diff
@@ -185,7 +205,7 @@ $EXPERIMENTAL
 struct Elements end
 OpticStyle(::Type{<:Elements}) = ModifyBased()
 
-function modify(f, obj, ::Elements)
+function modify(f, obj, ::Elements, object=List())
```

Comment:
There are other ideas what n argument modify could mean. I think defining it is not essential for this PR? So let's try to avoid it.

Reply:
Alternatively if you feel strongly about using `modify` we can also discuss that.

Reply:
The reason for adding this argument is so we can use the same code to both `Construct` new objects in and return tuples of fields, with `Tuple` (for all) and `Splat` (for some filtered subset). This is how Flatten.jl works, it's just impossible to read that that is whats happening - your structure is so much cleaner here.

We can also use a different method entirely to do this, this was just me trying to find the cleanest way of using everything already written here and build things so the other optics can use what Ive added.

Reply:
This method isn't working currently for that, it's just a dummy argument so it doesn't break.

Reply:
Ok sure, we can in the end decide whether to rename this. Competing ideas were to use extra args of modify for passing context around. Or have `modify(f, obj, lens)` behave to `map(f, itr)` as `modify(f, objs..., lens)` behaves to `map(f, itrs...)`.

Reply:
Better to keep the syntax open.

I've made them separate methods for now, we can work towards reducing the differences. Probably it should only affect the inner `_modify` not `modify` anyway.